src/rabbit_msg_store.erl

```diff
@@ -1400,17 +1404,36 @@ mark_handle_to_close(ClientRefs, FileHandlesEts, File, Invoke) ->
                  ets:match_object(FileHandlesEts, {{'_', File}, open}) ],
     true.
 
-safe_file_delete_fun(File, Dir, FileHandlesEts) ->
-    fun () -> safe_file_delete(File, Dir, FileHandlesEts) end.
+safe_file_delete_fun(File, Dir, FileHandlesEts, DiskErrorStrategy) ->
+    fun () -> 
+        safe_file_delete(File, Dir, FileHandlesEts, DiskErrorStrategy)
+    end.
 
-safe_file_delete(File, Dir, FileHandlesEts) ->
+safe_file_delete(File, Dir, FileHandlesEts, DiskErrorStrategy) ->
     %% do not match on any value - it's the absence of the row that
     %% indicates the client has really closed the file.
     case ets:match_object(FileHandlesEts, {{'_', File}, '_'}, 1) of
         {[_|_], _Cont} -> false;
-        _              -> ok = file:delete(
-                                 form_filename(Dir, filenum_to_name(File))),
-                          true
+        _              -> 
+            FileName = form_filename(Dir, filenum_to_name(File)),
+            case file:delete(FileName) of
+                ok -> ok;
+                {error, DeleteErr} ->
+                    rabbit_log:error(
+                        "Unable to delete message store file ~p, reason: ~p", 
+                        [FileName, DeleteErr]),
+                    case DiskErrorStrategy of
+                        ignore -> ok;
+                        crash -> throw({error, DeleteErr});
```

Comment:
Are you sure this will shut down the node instead of just the message store? The goal is to make nodes voluntarily stop/leave the cluster if there's a possibility of disk corruption.